akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala

```diff
@@ -100,6 +105,7 @@ import akka.serialization.Serializers
 
       val filtered = env.hasFiltered && env.getFiltered
       val source = if (env.hasSource) env.getSource else ""
+      val tags = env.getTagsList.iterator.asScala.toSet
```

Comment:
`if (env.hasTags)` or is that not needed for repeated?

Reply:
might also be good with an isEmpty check to optimize that case and avoid some conversions of empy list

Reply:
No hasNnn for repeated, but optimisation makes sense.